### DISCUSSION TITLE
Developer Guidelines

### DISCUSSION THREAD

**Discussion by poldi171254** [strawberrymusicplayer/strawberry#1396] [General]
The one thing I'm missing the most in Strawberry is a Remote capability.
I have over 1000 CD, collected over many years, and as we all know, not every song on every CD is great.
In Clementine, if I didn't like something I heard while I was away from the PC/Music System, I could just go Next.
I have added a basic Remote capability to Strawberry which will be available shortly for testing
I have also written a linux client to help in the testing and things are working as expected as far as I can tell.
The functionality is very basic and not like Clementine (no downloads, playlist changes, etc); all you can do is

Play, Pause, Next Previous, Stop

Both Strawberry and the Remote were build it Qt Creator; not sure if command line works.
When I started on this I knew nothing of QT, C++, Protobuf or CMake, so I'm sure the code can be improved. Last time I programmed anything in earnest was 20 years ago and that was in C.

 Still, I'd appreciate if anybody could let me know what the next steps are to get this approved/reviewed/tested.
 
 Thanks,
 Leo

**Comment by jonaski** [comment]
You can create a pull request so it can be reviewed, but it might take a long time, like mentioned in https://github.com/orgs/strawberrymusicplayer/discussions/682#discussioncomment-1192186 in 2021, I simply lack the time to dig into this, and the situation is even worse now, mainly because of the amount of support and upstream bugs and no other contributors to deal with issues. There are several bugs in both Qt and GStreamer that are not even reported upstream, we should at least take the time to report that, but we can't expect all gstreamer bugs to be fixed. Since the switch to playbin3 there are several stability issues affecting playback which we need to address. For me that is the main priority now, along with dealing with some design flaws inherited from Clementine.
I also like to review alternative ways of implementing a remote, particularity alternatives to protobuf. We really don't need protobuf for the tagreader anymore, so it could possibly be removed as dependency completely.
The best would be to use D-Bus, but the main issue with that is that you can't use d-bus over the network, so we need some network server to deal with that, but maybe MQTT or something else can be used for the server part.

**Comment by poldi171254** [comment]
Thanks for your feedback.
Given that I've spent the better part of 6 months getting my head around Qt, C++ & Strawberry, maybe I can help with the support.

I'll do a final review and create a pull request. I'll also provide a link to the remote code, which should make it easier to test the new functionality
